modules/certificate/certificate.service.js

```diff
@@ -67,17 +77,15 @@ class CertificatesService {
   }
 
   async generateCertificate(certificateData, userId, userRole) {
-    const randomNum = randomInt(100000, 999999);
-
-    certificateData.name = `hor${randomNum}`;
+    certificateData.name = await generateName();
 
     if (userRole === USER) {
       certificateData.ownedBy = userId;
     } else {
       certificateData.createdBy = userId;
     }
 
-    return new CertificateModel(certificateData).save();
+    return CertificateModel(certificateData).save();
```

Comment:
We need to invoke an emailsender if the email field has value.

Reply:
It will be in other service